### PR TITLE
feat(datepicker) : min mode, sets a lower limit for view mode (day, month, year)

### PR DESCRIPTION
**Pull request by rrambaud**
# PR Checklist
Before creating new PR, please take a look at checklist below to make sure that you've done everything that needs to be done before we can merge it.

 - [ x] read and followed the [CONTRIBUTING.md](https://github.com/valor-software/ngx-bootstrap/blob/development/CONTRIBUTING.md) guide.
 - [ x] built and tested the changes locally.
 - [ x] added/updated tests.
 - [ x] added/updated API documentation.
 - [ x] added/updated demos.

### CODE REVIEW

**Comment by rrambaud** [comment]
Created from feature request #2627 
Exists in AngularJS UI Bootstrap - http://angular-ui.github.io/bootstrap/versioned-docs/1.3.3/#/datepicker
Exists in ngx-bootstrap old datepicker

**Comment by rrambaud** [comment]
Created from feature request #2627 
Exists in AngularJS UI Bootstrap - http://angular-ui.github.io/bootstrap/versioned-docs/1.3.3/#/datepicker
Exists in ngx-bootstrap old datepicker

File: src/datepicker/reducer/bs-datepicker.reducer.ts [hunk 1]
```diff
@@ -48,14 +48,21 @@ export function bsDatepickerReducer(state = initialDatepickerState,
...
-
+      let newState;
+      let mode:BsDatepickerViewMode;
+      if (canSwitchMode(payload.viewMode, state.minMode)) {
+        mode = payload.viewMode;
+        newState = { view: { date, mode } };
+      }
+      else {
+        mode = state.view.mode
+        newState = { selectedDate: date, view: { date, mode } };
```
Comment (valorkin) [inline src/datepicker/reducer/bs-datepicker.reducer.ts:69]: not sure that selected date chance is necessary

**Comment by valorkin** [comment]
@rrambaud looks good to me overall, I will get to merge it a bit later

thanks for your help!

**Comment by YevheniiaMazur** [comment]
Has conflicting files. needs fix

**Comment by rrambaud** [comment]
I've fixed the conflicts

**Comment by ghost** [comment]
Hi,

I was wondering when this pull request can be finalized? We really need the possiblity to configure the datepicker with different modes.

Thanks,
Drazen

**Comment by rrambaud** [comment]
At least it's up to date

**Comment by ghost** [comment]
Please, is there any progress to expect soon? We need to figure out whether we will use this component or not.

Thanks

**Comment by ghost** [comment]
Hi, do you have any updates on this maybe?

Thanks,
Drazen

**Comment by Frennetix** [comment]
Any news on this?

**Comment by TreesAndMountains** [comment]
What's the status on this PR?

**Comment by ghost** [comment]
I think this PR will never be completed, looking at how much time it took so far

**Comment by valorkin** [comment]
Starting from monday full time dev will be assigned to ngx-bs, so things should get on speed again

**Comment by Frennetix** [comment]
So.. is it progressing?

**Comment by ghost** [comment]
Apparently not.

**Comment by bandhavya** [comment]
any updates on this?

**Comment by ducminhquan** [comment]
Any updates on this?

**Comment by djanjicek** [comment]
Come on guys, will this ever be completed?

**Comment by RookieLittleFatty** [comment]
hey, I don't know much about github, so i can't understand the PR and merge  about this feature.
Have your PR  been merged in any version? Can i use it now? Could you tell me how?  Thanks!

**Comment by mgpGit-zz** [comment]
What's the status on this feature? Is anyone working on it?

**Comment by santosmken** [comment]
What is the status of this feature?

**Comment by valorkin** [comment]
@domainv lets make n+1 release with most of datepicker PRs

**Comment by Domainv** [comment]
roger that

**Pull request merged**